packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3420,6 +3423,62 @@ AND event_name = '${eventName}'`,
     );
   }
 
+  //MKTODO: Explore how can I refactor getFactTableCTE to work for metrics or fact segments
+  private getFactTableCTEForSegments({
```

Comment:
@jdorn Any ideas on how to clean this up a bit? This is very similar to `getFactTableCTE` that takes in metrics.

I did explore combining this with the function above, but it turned into a case where so many of the function props had to be optional, requiring a lot of `if` checks.

Likewise, it got a bit complicated when building the `SQLVars`.